### PR TITLE
Vtgateclienttest

### PR DESCRIPTION
**Pull request by alainjobart**
@dumbunny this is to facilitate internal tests, and also add more coverage.

After import, I will remove the duplicate cut&paste stuff I had to add earlier.

### CODE REVIEW

File: py/vtdb/grpc_vtgate_client.py [hunk 1]
```diff
@@ -118,7 +116,8 @@ def rollback(self):
     finally:
       self.session = None
 
-  @vtgate_utils.exponential_backoff_retry((dbexceptions.TransientError))
+  @vtgate_utils.exponential_backoff_retry((dbexceptions.ThrottledError,
```
Comment (dumbunny) [inline py/vtdb/grpc_vtgate_client.py]: Exponential backoff on a ThrottledError sounds dangerous. I assume getting throttled and retrying is very cheap.

File: py/vtdb/vtgate_utils.py [hunk 2]
```diff
@@ -14,6 +14,12 @@
 BACKOFF_MULTIPLIER = 2
 
 
+# This pattern is used in transient error messages to differentiate
+# between a transient error and a throttling error.
+throttler_err_pattern = re.compile(
```
Comment (dumbunny) [inline py/vtdb/vtgate_utils.py]: When you compile a pattern, it's normally suffixed with "_re", not "_pattern".

File: py/vtdb/vtgate_client_testsuite.py [hunk 3]
```diff
@@ -0,0 +1,667 @@
...
+from vtdb import dbexceptions
+from vtdb import keyrange
+from vtdb import keyrange_constants
+from vtdb import vtgate_client
+from vtdb import vtgate_cursor
+from vtproto import query_pb2
+from vtproto import topodata_pb2
+
+
+class TestPythonClientBase(object):
```
Comment (dumbunny) [inline py/vtdb/vtgate_client_testsuite.py]: Good refactor.

File: py/vtdb/grpc_vtgate_client.py (hunk 1 above)
Comment (alainjobart) [inline py/vtdb/grpc_vtgate_client.py]: Yes it is. Just making this consistent though, that's what we do in other clients too.

File: py/vtdb/vtgate_utils.py (hunk 2 above)
Comment (alainjobart) [inline py/vtdb/vtgate_utils.py]: Done.

**Comment by alainjobart** [comment]
LGTM so I can merge.

**Comment by dumbunny** [comment]
LGTM

**Pull request merged**